mne/preprocessing/artifact_detection.py

```diff
@@ -4,12 +4,82 @@
 
 
 import numpy as np
-
 from ..annotations import (Annotations, _annotations_starts_stops)
 from ..transforms import (quat_to_rot, _average_quats, _angle_between_quats,
                           apply_trans, _quat_to_affine)
+from ..filter import filter_data
 from .. import Transform
-from ..utils import (_mask_to_onsets_offsets, logger)
+from ..utils import (_mask_to_onsets_offsets, logger, verbose)
+
+
+@verbose
+def annotate_muscle(raw, threshold=1.5, picks=None, min_length_good=.1,
+                    verbose=None):
+    """Detect segments with muscle artifacts.
+
+    Detects segments periods that contains high frequency activity beyond the
+    specified threshold. Muscle artifacts are most notable in the range of 110-
+    140Hz.
+
+    Raw data is band-pass filtered between 110 and 140 Hz, the signal envelope
+    computed, z-scored across samples, channel averaged and low-pass
+    filtered to smooth transient peaks.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to compute head position.
+    threshold : float
+        The threshold in z-scores for selecting segments with muscle activity
+        artifacts.
+    %(picks_all)s
+    min_length_good : int | float | None
+        The minimal good segment length between annotations, smaller segments
+        will be included in the movement annotation.
+    %(verbose)s
+
+    Returns
+    -------
+    annot : mne.Annotations
+        Periods with muscle artifacts annotated as BAD_muscle.
+    scores_muscle : array
+        Z-score values averaged across channels for each sample.
+    """
+    from scipy.stats import zscore
+    from scipy.ndimage.measurements import label
+
+    raw_copy = raw.copy()
+    raw_copy.pick(picks)
+    chan_type = raw_copy.get_channel_types()
```

Comment:
```suggestion
    ch_types = raw_copy.get_channel_types()
```

following convention elsewhere in the code